routers/web/auth/oauth.go

```diff
@@ -536,6 +536,16 @@ func AuthorizeOAuth(ctx *context.Context) {
 // GrantApplicationOAuth manages the post request submitted when a user grants access to an application
 func GrantApplicationOAuth(ctx *context.Context) {
 	form := web.GetForm(ctx).(*forms.GrantApplicationForm)
+
+	if !form.Granted {
+		handleAuthorizeError(ctx, AuthorizeError{
+			State:            ctx.Session.Get("state").(string),
```

Comment:
For `state`: it doesn't seem safe or right to do so.

Just ignore it? Or what does RFC say?